lib/gui/models/selection-state.js

```diff
@@ -78,10 +79,22 @@ SelectionStateModel.service('SelectionStateModel', function(DrivesModel) {
    * });
    */
   this.setImage = (image) => {
+    const previouslySelected = this.getDrive();
+
     Store.dispatch({
       type: Store.Actions.SELECT_IMAGE,
       data: image
     });
+
+    AnalyticsService.logEvent('Select image', image);
+
+    if (this.getDrive()) {
+      AnalyticsService.logEvent('Auto-selected drive', this.getDrive());
```

Comment:
This module shouldn't have the knowledge of how drives are auto-selected. Also, notice that a drive might get auto-selected even if no image is set, meaning that this will not catch that.

These events should live in the Redux store, which is where this logic is implemented. The Redux store is not AngularJS, so I guess let's leave this out from the PR and address it once we make the analytics module framework agnostic.